src/liballoc/collections/btree/node.rs

```diff
@@ -567,9 +571,11 @@ impl<'a, K: 'a, V: 'a, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
     }
 
     fn into_key_slice_mut(mut self) -> &'a mut [K] {
-        // Same as for `into_key_slice` above, we try to avoid a run-time check
-        // (the alignment comparison will usually be performed at compile-time).
-        if mem::align_of::<K>() > mem::align_of::<LeafNode<(), ()>>() && self.is_shared_root() {
+        // Same as for `into_key_slice` above, we try to avoid a run-time check.
```

Comment:
"(the alignment comparison will usually be performed at compile-time)." seemed a little vague (when not?) and probably applies to `into_key_slice` too.

Reply:
"the alignment and size comparison should always be performed at compile-time"

When not? When the optimizer is silly. It really should always constant-fold this.